src/mapml/layers/FeatureIndex.js

```diff
@@ -0,0 +1,147 @@
+                    body.appendChild(this._updateOutput("More results", 0, 9));
+                }
+                index += 1;
+            }
+        });
+        this._addToggleKeys();
+    },
+
+    _updateOutput: function (label, index, key) {
+        let span = document.createElement("span");
+        span.setAttribute("index", index);
+        span.innerHTML = `<kbd>${key}</kbd>` + " " + label;
+        return span;
+    },
+
+    _addToggleKeys: function () {
+        let allFeatures = this._body.allFeatures;
+        for(let i = 0; i < allFeatures.length; i++){
+            if(allFeatures[i].length === 0) return;
+            if(allFeatures[i - 1]){
+                let label = "Previous results";
+                allFeatures[i].push({label});
+            }
+
+            if(allFeatures[i + 1] && allFeatures[i + 1].length > 0){
+                let label = "More results";
+                allFeatures[i].push({label});
+            }
+        }
+    },
+
+    _toggleContent: function (e){
+        let body = this._body;
+        if(e.originalEvent.keyCode === 57){
+            this._newContent(body, 1);
+        } else if(e.originalEvent.keyCode === 56){
+            this._newContent(body, -1);
+        }
+    },
+
+    _newContent: function (body, direction) {
+        let index = body.firstChild.getAttribute("index");
+        let newContent = body.allFeatures[Math.floor(((index - 1) / 7) + direction)];
+        if(newContent && newContent.length > 0){
+            body.innerHTML = "";
+            for(let i = 0; i < newContent.length; i++){
+                let feature = newContent[i];
+                let index = feature.index ? feature.index : 0;
+                let key = i + 1;
+                if (feature.label === "More results") key = 9;
+                if (feature.label === "Previous results") key = 8;
+                body.appendChild(this._updateOutput(feature.label, index, key));
+            }
+        }
+    },
+
+    _toggleEvents: function (){
+        this._map.on("viewreset move moveend focus blur", this._addOrRemoveFeatureIndex, this);
+
+    },
+
+    _addOrRemoveFeatureIndex: function (e) {
+        let obj = this;
+        setTimeout(function() {
+            if (e && e.type === "focus") {
+                obj._container.querySelector('rect').style.display = "inline";
+                obj._output.style.display = "block";
+            } else if (e && e.type === "blur") {
+                obj._container.querySelector('rect').style.display = "none";
+                obj._output.style.display = "none";
+            } else if (obj._map.isFocused) {
+                obj._container.querySelector('rect').style.display = "inline";
+                obj._output.style.display = "block";
+            } else {
+                obj._container.querySelector('rect').style.display = "none";
+                obj._output.style.display = "none";
+            }
```

Comment:
```suggestion
            if (e && e.type === "focus") {
                obj._container.removeAttribute("hidden");
                obj._output.removeAttribute("hidden");
            } else if (e && e.type === "blur") {
                obj._container.setAttribute("hidden", "");
                obj._output.setAttribute("hidden", "");
            } else if (obj._map.isFocused) {
                obj._container.removeAttribute("hidden");
                obj._output.removeAttribute("hidden");
            } else {
                obj._container.setAttribute("hidden", "");
                obj._output.setAttribute("hidden", "");
            }
```

The `rect` selectors need to change per the [previous suggestion](https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/636/files#r771174496). This change also removes inline styles (per motivation linked in that comment).

I changed to hide the entire `mapml-feature-index-box` and not only the SVG within it, though we can change to only hide the SVG if there's a reason not to hide the parent container.